### PR TITLE
Use parquet type for airdrops

### PR DESCRIPTION
**Pull request by yabirgb**
[skip ci]

Closes #(issue_number)

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.

### CODE REVIEW

**Comment by LefterisJP** [comment]
why skip ci?

**Comment by yabirgb** [comment]
Because I need to update the remote index to change .csv for .parquet and also the tests with the same change

File: rotkehlchen/chain/ethereum/airdrops.py [hunk 1]
```diff
@@ -471,7 +459,6 @@ def process_airdrop_with_api_data(
 
 
 def _process_csv_airdrop(
-        msg_aggregator: MessagesAggregator,
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/airdrops.py:474]: you would need to change the method name and docstring of this

File: rotkehlchen/chain/ethereum/airdrops.py [hunk 2]
```diff
@@ -501,13 +488,15 @@ def _process_csv_airdrop(
...
     temp_found_data: dict[ChecksumEvmAddress, dict] = defaultdict(lambda: defaultdict(dict))
     temp_airdrop_tuples = []
-    for row in get_airdrop_data(airdrop_data, protocol_name, data_dir):
-        if len(row) < 2:
-            msg_aggregator.add_warning(f'Skipping airdrop CSV for {protocol_name} because it contains an invalid row: {row}')  # noqa: E501
-            break
-        addr, amount, *_ = row
-        # not doing to_checksum_address() here since the file addresses are checksummed
-        # and doing to_checksum_address() so many times hits performance
+    airdrop_lazy_df = get_airdrop_data(airdrop_data, protocol_name, data_dir)
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/airdrops.py:491]: better name. I don't know what a df is

```suggestion
    airdrop_lazy_df_make_no_sense = get_airdrop_data(airdrop_data, protocol_name, data_dir)
```

**Review by LefterisJP — COMMENTED** [review]
you would also need to make sure all unnecessary csv files in the user's local are deleted

File: rotkehlchen/chain/ethereum/airdrops.py (hunk 2 above)
Comment (yabirgb) [reply rotkehlchen/chain/ethereum/airdrops.py:491]: dataframe. https://docs.pola.rs/api/python/stable/reference/dataframe/index.html 
imagen

File: rotkehlchen/chain/ethereum/airdrops.py [hunk 3]
```diff
@@ -473,15 +462,14 @@ def process_airdrop_with_api_data(
...
+def _process_airdrop_file(
         addresses: Sequence[ChecksumEvmAddress],
         data_dir: Path,
-        airdrop_data: CSVAirdrop,
+        airdrop_data: AirdropFileMetadata,
         protocol_name: str,
         tolerance_for_amount_check: FVal,
 ) -> tuple[list[AirdropClaimEventQueryParams], dict[ChecksumEvmAddress, dict]]:
-    """Process csv to find the addresses that have a claimable amount.
+    """Process airdrop's file to find the addresses that have a claimable amount.
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/airdrops.py:472]: either

```suggestion
    """Process airdrop file to find the addresses that have a claimable amount.
```
or

```suggestion
    """Process an airdrop's file to find the addresses that have a claimable amount.
```

File: rotkehlchen/db/upgrades/v42_v43.py [hunk 4]
```diff
@@ -44,6 +46,15 @@ def _remove_coinbasepro_credentials(write_cursor: 'DBCursor') -> None:
     )
 
 
+@enter_exit_debug_log()
+def _remove_old_csv_files(data_dir: Path) -> None:
+    """Delete old csv files for airdrops"""
+    airdrops_dir = data_dir / APPDIR_NAME / AIRDROPSDIR_NAME
+    for csv_file_path in airdrops_dir.glob('*.csv'):
+        csv_file_path.unlink()
```
Comment (LefterisJP) [inline rotkehlchen/db/upgrades/v42_v43.py:54]: This is really sensitive code here. Please handle any unlink errors, as they would crash the app mid-upgrade and that would be bad.

File: rotkehlchen/tests/db/test_db_upgrades.py [hunk 5]
```diff
@@ -2377,6 +2381,18 @@ def test_upgrade_db_42_to_43(user_data_dir, messages_aggregator):
             ('coinbasepro', Location.COINBASEPRO.serialize_for_db(), 'api_key', 'api_secret', 'passphrase'),  # noqa: E501
         ).rowcount == 1
 
+    # create csv files that will be deleted in the db upgrade and one that will be kept
```
Comment (LefterisJP) [inline rotkehlchen/tests/db/test_db_upgrades.py:2384]:
```suggestion
    # create csv files that will be deleted in the db upgrade and a parquet one to keep
```

**Review by LefterisJP — COMMENTED** [review]
Some smaller comments and error handling.

Also questions.
1. You tested this and it works right?
2. When converting to parquet files have you made sure all addresses are checksummed?

**Comment by yabirgb** [comment]
1. Yes, tested it and works fine
2. Yes, we kept the same checks in the data repo https://github.com/rotki/data/blob/73c11e4aa56639978bf3b6d272128dcb6877d31a/tests/test_airdrops.py#L46

**Review by LefterisJP — APPROVED** [review]
LGTM

**Pull request merged**